### PR TITLE
[feature] Return NotFound for Nonexistent Preprint Providers [OSF-7621]

### PR DESCRIPTION
**Pull request by caseyrollins**
#### Purpose
- Currently, the API incorrectly returns a `200` for requests to an invalid preprint provider, and `502`s for every sub-endpoint request to a fake provider.
  - `200`: https://api.osf.io/v2/preprint_providers/fake/
  - `502`: https://api.osf.io/v2/preprint_providers/fake/licenses/ 
  - `502`: https://api.osf.io/v2/preprint_providers/fake/preprints/ 
  - `502`: https://api.osf.io/v2/preprint_providers/fake/taxonomies/ 
- This PR checks that the requested preprint provider exists for all endpoints listed above and appropriately returns a `404` if the provider does not exist. 

#### Changes
- `PreprintProvider.load('fake')` --> `get_object_or_error(PreprintProvider, 'fake')`

#### Side effects
- Less fun pretending https://api.osf.io/v2/preprint_providers/stop_trying_to_make_fetch_happen could be a real provider.

#### Ticket
- [OSF-7621](https://openscience.atlassian.net/browse/OSF-7621)

### CODE REVIEW

File: api_tests/preprint_providers/views/test_preprint_provider_subjects_list.py [hunk 1]
```diff
@@ -6,6 +6,28 @@
 from osf_tests.factories import SubjectFactory, PreprintProviderFactory
 
 
+class TestPreprintProviderExists(ApiTestCase):
+
+    # Regression for https://openscience.atlassian.net/browse/OSF-7621
+
+    def setUp(self):
+        super(TestPreprintProviderExists, self).setUp()
+        self.url = '/{}/preprint_proiders/fake/'.format(API_BASE)
```
Comment (alexschiller) [inline api_tests/preprint_providers/views/test_preprint_provider_subjects_list.py]: Typo (`preprint_proiders`) which will always 404, can you fix the typo and add a passing test for sanity sake?

Comment (caseyrollins) [inline api_tests/preprint_providers/views/test_preprint_provider_subjects_list.py]: Ah, nice catch. 

Gonna move into a new file as well. I didn't realize I added this test in `test_preprint_provider_subjects_list.py`, where it doesn't really belong.

**Comment by brianjgeiger** [comment]
@alexschiller Merge conflict

**Comment by brianjgeiger** [comment]
@caseyrollins Merge conflict

**Pull request merged**